for-research/public-likelihoods/public-likelihoods.ipynb

```diff
@@ -0,0 +1,936 @@
+    "ROOT.gStyle.SetOptStat(False) # turn off the stats-box that ROOT will display by default on every plot\n",
+    "c = ROOT.TCanvas(\"canv\",\"canv\",900,900)\n",
+    "ROOT.gStyle.SetLegendFont(43);ROOT.gStyle.SetLegendTextSize(12);\n",
+    "w[\"models/simPdf\"].Draw(\"eratio\") # draw model with error bars and ratio plot\n",
+    "c.Draw()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb976100-622b-4141-b822-d0fe17342559",
+   "metadata": {},
+   "source": [
+    "# 2. Modifying workspaces\n",
+    "\n",
+    "Workspaces can be modified 'in-situ' or a new workspace can be created and objects added to it. We will see an example of both of these now ...\n",
+    "\n",
+    "## 2.1. Modifying in-situ: Reparameterizing a parameter\n",
+    "\n",
+    "We can use the `xRooNode::Replace` method to replace any variable with a generic function. The function must follow RooFit's [WS-factory language](https://root.cern.ch/doc/master/classRooFactoryWSTool.html#a228d9cb336a3bdc7dfe260f4c6aaf8df). For example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "1f9695ee-06d0-4fdd-88da-e6f0c4ea98e1",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "xRooNode{}"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Warning in <xRooNode::Replace>: Replacing dPhi_ttbar_Z_Bin_001_mu in all clients: dPhi_ttbar_Z_Bin_005_mu,mu_prime_2,model_reg_3l_WZ_dPhi_ttbar_Z_reg_3l_WZ_Truth_bin_1_scaleFactors,model_reg_3l_tZq_dPhi_ttbar_Z_reg_3l_tZq_Truth_bin_1_scaleFactors,model_reg_4l_DF_dPhi_ttbar_Z_reg_4l_DF_Truth_bin_1_scaleFactors,model_reg_3l_ttZ_dPhi_ttbar_Z_reg_3l_ttZ_Truth_bin_1_scaleFactors,model_reg_4l_SF_dPhi_ttbar_Z_reg_4l_SF_Truth_bin_1_scaleFactors,model_reg_4l_ZZ_dPhi_ttbar_Z_reg_4l_ZZ_Truth_bin_1_scaleFactors,\n",
+      "Info in <xRooNode::SaveAs>: workspace saved to reparameterized_ws.root\n",
+      "Warning in <xRooNode::Replace>: Replacing myFunc in all clients: dPhi_ttbar_Z_Bin_005_mu,mu_prime_2,model_reg_3l_WZ_dPhi_ttbar_Z_reg_3l_WZ_Truth_bin_1_scaleFactors,model_reg_3l_tZq_dPhi_ttbar_Z_reg_3l_tZq_Truth_bin_1_scaleFactors,model_reg_4l_DF_dPhi_ttbar_Z_reg_4l_DF_Truth_bin_1_scaleFactors,model_reg_3l_ttZ_dPhi_ttbar_Z_reg_3l_ttZ_Truth_bin_1_scaleFactors,model_reg_4l_SF_dPhi_ttbar_Z_reg_4l_SF_Truth_bin_1_scaleFactors,model_reg_4l_ZZ_dPhi_ttbar_Z_reg_4l_ZZ_Truth_bin_1_scaleFactors,\n"
+     ]
+    }
+   ],
+   "source": [
+    "poiName = w.poi()[0].GetName() # get the name of the first POI\n",
+    "w.poi()[0].Replace(f\"expr::myFunc('2*{poiName}',{poiName})\") # example of replacing the parameter everywhere in the workspace with a function that multiplies it by 2\n",
+    "\n",
+    "# we can save this modified workspace like this:\n",
```

Comment:
Maybe "into a ROOT file"? This could catch some people off-guard, as you initially loaded the thing from a json file, so they might not be expecting ROOT I/O here.